.github/workflows/cleanup.yml

```diff
@@ -280,12 +288,12 @@ jobs:
             await github.rest.issues.createComment({
               owner: context.repo.owner,
               repo: context.repo.repo,
-              issue_number: ${{ github.event.pull_request.number }},
+              issue_number: ${{ inputs.pr_number }},
               body: `**\`Access on Demand\`** - 🟦 **\`IAM\`** request not found, skip cleanup.`,
             });
       - name: 'Delete Branch'
         # Only delete branch when cleanups are skipped or successfully handled.
-        if: '${{ always() && steps.cleanup_iam.outcome != ''failure'' && steps.cleanup_tool.outcome != ''failure'' }}'
```

Comment:
Removed always() as we should't delete the branch if any of the previous steps failed.